packages/agents-copilotstudio-client/src/copilotStudioWebChat.ts

```diff
@@ -0,0 +1,164 @@
+   * @returns An observable that emits the activity ID.
+   */
+  postActivity(activity: Activity): Observable<string>;
+
+  /**
+   * Ends the connection.
+   * This will complete the connectionStatus$ and activity$ observables.
+   */
+  end(): void;
+}
+
+/**
+ * This class is intended to be used in WebChat applications to connect to the Copilot Studio service.
+ *
+ * example usage:
+ * ```javascript
+ * const client = new window.Agents.CopilotStudioClient(...)
+ * window.WebChat.renderWebChat({
+ *   directLine: window.Agents.CopilotStudioWebChat.createConnection(client)
+ * })
+ * ```
+ */
+export class CopilotStudioWebChat {
+  /**
+   * Creates a new DirectLine-Like connection to WebChat.
+   * When an activity is posted in WebChat, the connection will send it to the Copilot Studio service, awaiting response.
+   * After a response is received, it will emit the incoming activity back to WebChat.
+   *
+   * @param client - The Copilot Studio client instance.
+   * @param settings - Optional settings for the WebChat connection.
+   * @returns A new instance of CopilotStudioWebChatConnection.
+   */
+  static createConnection (
+    client: CopilotStudioClient,
+    settings?: CopilotStudioWebChatSettings
+  ):CopilotStudioWebChatConnection {
+    let sequence = 0
+    let activityObserver: Observer<Partial<Activity>> | undefined
+    let conversation: ConversationAccount | undefined
+
+    const connectionStatus$ = new BehaviorSubject(0)
+    const activity$ = Observable.create(
+      async (observer: Observer<Partial<Activity>>) => {
+        activityObserver = observer
+
+        if (connectionStatus$.value < 2) {
+          connectionStatus$.next(2)
+          return
+        }
+
+        notifyTyping()
+        const activity = await client.startConversationAsync()
+        conversation = activity.conversation
+        sequence = 0
+        notifyActivity(activity)
+      }
+    )
+
+    const notifyActivity = (activity: Partial<Activity>) => {
+      activityObserver?.next({
+        ...activity,
+        timestamp: new Date().toISOString(),
+        channelData: {
+          ...activity.channelData,
+          'webchat:sequence-id': sequence++,
+        },
+      })
+    }
+
+    const notifyTyping = () => {
+      if (!settings?.showTyping) {
+        return
+      }
+
+      const from = conversation
+        ? { id: conversation.id, name: conversation.name }
+        : { id: 'agent', name: 'Agent' }
+      notifyActivity({ type: 'typing', from })
+    }
+
+    return {
+      connectionStatus$,
+      activity$,
+      postActivity (activity: Activity) {
+        if (!activity.text?.trim()) {
+          throw new Error('Activity text cannot be empty.')
+        }
+
+        if (!activityObserver) {
+          throw new Error('Activity observer is not initialized.')
+        }
+
+        return Observable.create(async (observer: Observer<string>) => {
+          try {
+            const id = crypto.randomUUID()
```

Comment:
we are using uuid to generate guids, and maybe we could avoid the `crypto` dependency.